File: src/Model/Expression/Expression.java
package Model.Expression;

import DataTypes.MyIDictionary;
import DataTypes.MyIHeap;
import Exception.ExpressionEvaluationException;

import java.io.Serializable;

/**
 * Created by vladc on 12.10.2016.
 */
public abstract class Expression implements Serializable {
    public abstract int evaluate(MyIDictionary<String, Integer> symTab, MyIHeap heap) throws ExpressionEvaluationException;
    public abstract String toString();
}

File: Toy-Language-Interpreter_C#/Toy-Language-Interpreter/Model/Expression/Expression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Toy_Language_Interpreter.DataTypes;

namespace Toy_Language_Interpreter.Model.Expression
{
    abstract class AExpression
    {
        public abstract int Evaluate(MyIDictionary<string, int> symTab);
        public override abstract string ToString();
    }
}
